cpp/CMakeLists.txt

```
@@ -55,99 +64,95 @@ if(KvikIO_REMOTE_SUPPORT)
+    set(cuFile_BATCH_API_FOUND TRUE)
   endif()
-
-  include(cmake/thirdparty/get_nvtx.cmake)
+  message(STATUS "Found cuFile Batch API: ${cuFile_BATCH_API_FOUND}")
+  string(FIND "${CUFILE_H_STR}" "cuFileReadAsync" cuFileReadAsync_location)
+  if(cuFileReadAsync_location EQUAL "-1")
+    set(cuFile_STREAM_API_FOUND FALSE)
+  else()
+    set(cuFile_STREAM_API_FOUND TRUE)
+  endif()
+  message(STATUS "Found cuFile Stream API: ${cuFile_STREAM_API_FOUND}")
 endif()
 
 include(cmake/thirdparty/get_thread_pool.cmake)
 
-# library targets
-add_library(kvikio INTERFACE)
-add_library(kvikio::kvikio ALIAS kvikio)
+# ##################################################################################################
+# * library targets --------------------------------------------------------------------------------
 
-# We enable CUDA and cuFile both here and in the FINAL_CODE_BLOCK export block. While the code block
-# below (in FINAL_CODE_BLOCK) sets this information when KvikIO is imported from a
-# kvikio-config.cmake file, this code block is intended to be used by projects that include KvikIO's
-# source directory in their own CMake build.
-#
-# Normally we would just set the below without using $<BUILD_LOCAL_INTERFACE:...>, and without the
-# final_code_string, but in this case we want to conditionally set these things at import time, not
-# export time, since KvikIO is a header-only library that can adapt to different build environments.
-
-# Enable CUDA in KvikIO
-if(CUDAToolkit_FOUND)
-  if(CUDA_STATIC_RUNTIME)
-    target_link_libraries(kvikio INTERFACE $<BUILD_LOCAL_INTERFACE:CUDA::cudart_static>)
-  else()
-    target_link_libraries(kvikio INTERFACE $<BUILD_LOCAL_INTERFACE:CUDA::cudart>)
-  endif()
-  target_compile_definitions(kvikio INTERFACE $<BUILD_LOCAL_INTERFACE:KVIKIO_CUDA_FOUND>)
-else()
-  message(WARNING "Building KvikIO without CUDA")
-endif()
+file(GLOB SOURCES "src/*.cpp")
+add_library(kvikio ${SOURCES})
 
-# Enable supported cuFile features in KvikIO
-if(cuFile_FOUND)
-  target_link_libraries(kvikio INTERFACE $<BUILD_LOCAL_INTERFACE:cufile::cuFile_interface>)
-  target_compile_definitions(kvikio INTERFACE $<BUILD_LOCAL_INTERFACE:KVIKIO_CUFILE_FOUND>)
-  if(cuFile_BATCH_API_FOUND)
-    target_compile_definitions(
-      kvikio INTERFACE $<BUILD_LOCAL_INTERFACE:KVIKIO_CUFILE_BATCH_API_FOUND>
-    )
-  endif()
-  if(cuFile_STREAM_API_FOUND)
-    target_compile_definitions(
-      kvikio INTERFACE $<BUILD_LOCAL_INTERFACE:KVIKIO_CUFILE_STREAM_API_FOUND>
-    )
-  endif()
-endif()
+# To avoid symbol conflicts when statically linking to libcurl.a (see get_libcurl.cmake) and its
+# dependency OpenSSL, we exclude them when building libkvikio.so. This way, libkvikio.so will not
+# expose any OpenSSL symbols that could conflict with downstream users like CPython that also links
+# to (another version of) OpenSSL.
+target_link_options(kvikio PRIVATE "-Wl,--exclude-libs,ALL")
```

```suggestion
target_link_options(kvikio PRIVATE "LINKER:--exclude-libs,ALL")
```
Allow CMake to convert `LINKER:` to `-Wl,` or `-Xlinker` depending on the compiler toolchain used